sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusConnectionStringProperties.java

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.amqp.implementation.ConnectionStringProperties;
+
+import java.net.URI;
+
+/**
+ * A utility class that parses a connection string into sections.
+ */
+public final class ServiceBusConnectionStringProperties {
+    private final URI endpoint;
+    private final String entityPath;
+    private final String sharedAccessKeyName;
+    private final String sharedAccessKey;
+    private final String sharedAccessSignature;
+
+    private ServiceBusConnectionStringProperties(ConnectionStringProperties properties) {
+        this.endpoint = properties.getEndpoint();
+        this.entityPath = properties.getEntityPath();
+        this.sharedAccessKeyName = properties.getSharedAccessKeyName();
+        this.sharedAccessKey = properties.getSharedAccessKey();
+        this.sharedAccessSignature = properties.getSharedAccessSignature();
+    }
+
+    /**
+     * Parse a ServiceBus connection string into an instance of this class.
+     * @param connectionString The connection string to be parsed.
+     * @return An instance of this class.
+     */
+    public static ServiceBusConnectionStringProperties parse(String connectionString) {
+        return new ServiceBusConnectionStringProperties(new ConnectionStringProperties(connectionString));
```

Comment:
Update javadoc to include the exceptions thrown if the connection string is invalid.

Reply:
Added